apiserver/common/secrets/drain.go

```diff
@@ -62,44 +62,28 @@ func NewSecretsDrainAPI(
 }
 
 // GetSecretsToDrain returns metadata for the secrets that need to be drained.
-func (s *SecretsDrainAPI) GetSecretsToDrain(ctx context.Context) (params.ListSecretResults, error) {
+func (s *SecretsDrainAPI) GetSecretsToDrain(ctx context.Context) (params.SecretRevisionsToDrainResults, error) {
 	var (
-		metadata         []*coresecrets.SecretMetadata
-		revisionMetadata [][]*coresecrets.SecretRevisionMetadata
-		err              error
+		toDrain []*coresecrets.SecretMetadataForDrain
+		err     error
 	)
 	if s.authTag.Kind() == names.ModelTagKind {
-		metadata, revisionMetadata, err = s.secretService.ListUserSecrets(ctx)
+		toDrain, err = s.secretService.ListCharmSecretsToDrain(ctx)
```

Comment:
Drain is for user secrets only.

Reply:
Oh, is a typo. There is a ListUserSecretsToDrain method. Fixed and added a test